src/cls/IPM/Repo/Oras/PackageService.cls

```diff
@@ -102,6 +102,11 @@ Method ListModulesFromTagString(allTagsString As %String, semverExpr As %IPM.Gen
 			Continue
 		}
 
+		#; Special case: if we provided an explicit build number, require it.
+		If ##class(%IPM.General.SemanticVersion).IsValid(searchCriteria.VersionExpression,.specificVersion) && (specificVersion.Build '= "") && (specificVersion.Build '= tVersion.Build) {
```

Comment:
Could some comments be added to the method for the expected format of allTagsString, what type of objects moduleList contains and what is name here?